### PR TITLE
Docs updates

### PR DESCRIPTION
**Pull request by jinglin-l**
This PR updates the setup documentation to include new package managers (uv and bun) and clarifies installation steps for better organization and clarity

### CODE REVIEW

**Comment by Theodlz** [comment]
@jinglin-l I think you forgot to install the pre-commit hook (as per the dev section of the docs), so the files are not formatted correctly. Could you fix that please?

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md]: so uv and bun are package managers but also virtualenv & javascript runtimes respectively. Maybe you can have one section for each as they are not just package managers. One section for each is also a nice way to explain how we get the python part to work, and the js part. Might want to have some little text there saying that the python part is for the backend, and js for the frontend. Could also be still one section but with 2 sub sections?

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md]: let's not even put instructions for fish shell honestly

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md]: many of our installation steps require that. I'd just tell people to reload their shell or open a new terminal and they can figure it out.

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md]: Nice, that's a great idea

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md]: You can delete that entirely, we don't need that section anymore

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md:115]: here there is a mention of <version> but it's not in the actual template underneath, can you add it to the template? thanks

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md]: I don't think this shows the version number that's in the path, right? If not, please explain how to get it (with ls)

Comment (Theodlz) [reply doc/setup.md]: I would replace "have forgotten" by "did not take note of"

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md:150]: why the modifications here?

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md]: can remove the conda instructions

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md]: changes should be made to config.yaml (copied from the default), never to the default file itself!

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md]: I think it's not super clear that this should be edited, rather than just added to the end of the file. Maybe just a wording thing, but adding some `...` here and there and/or comments to make it clear in that template that you are basically showing locations in the whole file and not just something to copy/paste would be super nice

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md]: No very helpful, at least as is. You can remove that

**Review by Theodlz — CHANGES_REQUESTED** [review]
I left a first batch of comments, I think we're very close to have something we can merge. I'm making changes to the Dockerfile that uses uv and bun at the moment, so we might be able to merge my PR too next week if yours is ready soon!

File: doc/setup.md
Comment (jinglin-l) [reply doc/setup.md]: Ok! I like the idea of one section, perhaps titled `Development Setup`, and then sub sections for BE/FE.

File: doc/setup.md
Comment (jinglin-l) [reply doc/setup.md]: For me, brew info does show an install path with a version. Do you get some other output? 

```
jinglinli@infra04-wg012 skyportal % brew info hdf5
==> hdf5: stable 1.14.4.3 (bottled)
File format designed to store large amounts of data
https://www.hdfgroup.org/solutions/hdf5/
Conflicts with:
  hdf5-mpi (because hdf5-mpi is a variant of hdf5, one can only use one or the other)
Installed
**/opt/homebrew/Cellar/hdf5/1.14.4.3 (275 files, 21.6MB) ***
  Poured from bottle using the formulae.brew.sh API on 2024-10-06 at 23:50:23
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/h/hdf5.rb
License: BSD-3-Clause
==> Dependencies
Build: cmake ✘
Required: gcc ✘, libaec ✔, pkg-config ✔
==> Analytics
install: 22,318 (30 days), 59,405 (90 days), 214,990 (365 days)
install-on-request: 12,084 (30 days), 29,304 (90 days), 110,883 (365 days)
build-error: 14 (30 days)

```

File: doc/setup.md
Comment (jinglin-l) [reply doc/setup.md:150]: I think this may have been a mistake. I'll revert this change.

File: doc/setup.md
Comment (jinglin-l) [reply doc/setup.md]: True! Oopsie. Thanks for catching this! 😄

File: doc/setup.md
Comment (jinglin-l) [reply doc/setup.md]: I'll add some ellipses to clarify this intent!

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md]: No need for windows, we definitely don't support nor do most server-side applications. We don't and won't expect to run SkyPortal there. You could just remove that windows line

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md]: Might not want to say "development" here, this is just the setup to run the app locally, not necessarily to do any dev

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md]: Same comment here about Windows support

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md]: Did you leave something like this for uv? Might be some git diff weirdness but while comparing with the previous version, it looks as if you only show that command for bun now

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md:149]: Why this addition, did you need that to run it?

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md]: I'd remove the `skyportal/` from the path here. Just to avoid confusing because we have a skyportal directory in the skyportal repo, so folks might look inside of this one instead of just looking in the top directory. That's a very minor concern, so you can also leave that there

File: doc/setup.md
Comment (Theodlz) [inline doc/setup.md:243]: Nice, thanks!

**Review by Theodlz — CHANGES_REQUESTED** [review]
Some very minor comments! Once addressed let's merge this

**Comment by Theodlz** [comment]
@jinglin-l any updates?

File: doc/setup.md
Comment (jinglin-l) [reply doc/setup.md]: True, I'll remove it.

File: doc/setup.md
Comment (jinglin-l) [reply doc/setup.md:149]: I'm adding back a line I accidentally removed in a previous commit. The formatting changed a bit which is why it looks like an addition/change in the git diff.

**Comment by jinglin-l** [comment]
> @jinglin-l any updates?

Just pushed! Feel free to review.

**Review by Theodlz — APPROVED** [review]
LGTM! Excellent work @jinglin-l

**Comment by Theodlz** [comment]
I'll let you merge it :) @jinglin-l

**Pull request merged**
